lib/agent.js

```diff
@@ -41,10 +46,11 @@ class Agent extends EventEmitter {
     let pool = self[kAgentCache].get(origin)
 
     if (!pool) {
-      pool = self[kAgentOpts] && self[kAgentOpts].connections === 1
-        ? new Client(origin, self[kAgentOpts])
-        : new Pool(origin, self[kAgentOpts])
+      const Klass = self[kAgentOpts] && self[kAgentOpts].connections === 1
+        ? self[kClientClass]
+        : self[kPoolClass]
 
+      pool = new Klass(origin, self[kAgentOpts])
```

Comment:
Use a factory function instead as argument, i.e.

```js
pool = factory(origin, self[kAgentOpts])
```

Reply:
Nice, I really like it. It will also simplify the Typescript definitions.
I just have a question: do I also need a RedirectClient or it is perfectly fine to always instantiate a subclass of `Pool` even if `connections==1`?

Reply:
You don't need redirect client. It's just a micro optimization.

Reply:
Done!